File: reverse.py
import random


def reverse(s):
    if len(s) == 1:
        return s
    else:
        return reverse(s[1:]) + s[0]


def test(n):
    for i in range(n):
        s = ''.join(random.sample('asdjbfnjasenlfj认得你飞机库fnerkfnini分矿务局dskfnk', 8))
        assert reverse(s) == ''.join(list(reversed(s))), '测试失败'


if __name__ == '__main__':
    test(random.randint(1, 100))
    print(reverse(input('请输入字符串\n')))

File: reverse.java
// 编写一个递归版本的 reverse(s) 函数(或方法),以将字符串s倒置。
public class reverse {
    // 递归
    public static String reversed(String str){
        int length = str.length();
        if (length <= 1) {
            return str;
        }
        String left = str.substring(0, length / 2);
        String right = str.substring(length/2, length);
        return reversed(right) + reversed(left);
    }
    public static void main(String[] args) {
        String str = "acbdfqe";
        String reversed = reversed(str);
        System.out.println(reversed);
    }
}
